libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -297,119 +297,26 @@ class LC_CSR_CSC_Labeled_Graph
-        // Update bounds for last only if we're searching for the first element.
-        if (finding_first) {
-          last_begin_bound = mid;
-          last_end_bound = r + 1;
-        }
-
-        bool mid_within_limit = finding_first ? mid > limit : mid < limit;
-        if (!(opt == FindEdgeWithLabelOptions::FindAny) && mid_within_limit) {
-          // check that mid - 1 is not key.
-          GraphNode adjacent_value = in_edges
-                                         ? this->getInEdgeDst(mid + direction)
-                                         : this->getEdgeDst(mid + direction);
-          if (adjacent_value != key) {
-            return std::make_tuple(mid, last_begin_bound, last_end_bound);
-          } else {
-            if (finding_first) {
-              r = mid - 1;
-            } else {
-              l = mid + 1;
-            }
-          }
-        } else {
-          return std::make_tuple(mid, last_begin_bound, last_end_bound);
-        }
-      }
-      if (value < key)
-        l = mid + 1;
-      else
-        r = mid - 1;
-    }
-    return std::nullopt;
-  }
-
-  /**
-   * Returns an edge iterator to an edge with some node and key with some label
-   * by searching for the key via the node's outgoing or incoming edges. By
-   * supplying find_first or find_last we can find the first or last satisfying
-   * edge. If neither are provided we will return an arbitrary satisfying edge.
-   *
-   * The return value is a tuple of <edge, lower bound for searching for last
-   * element, upper bound for searching for last element> where the lower bound
-   * is inclusive and the upperbound is exclusive.
-   *
-   * if opt != FindFirst, the second two elements are arbitrary.
-   *
-   * If not found, returns nothing.
-   */
-  template <bool in_edges, enum FindEdgeWithLabelOptions opt>
-  std::optional<std::tuple<edge_iterator, edge_iterator, edge_iterator>>
-  FindFirstOrLastEdgeWithLabel(
-      GraphNode node, GraphNode key, const EdgeTy& data) const {
-    edge_iterator begin =
-        in_edges ? in_raw_begin(node, data) : raw_begin(node, data);
-    edge_iterator end = in_edges ? in_raw_end(node, data) : raw_end(node, data);
-    return FindFirstOrLastEdge<in_edges, opt>(begin, end, key);
-  }
-
-  /**
-   * Returns an edge iterator to an edge with some node and key with some label
-   * by searching for the key via the node's outgoing or incoming edges.
-   * If not found, returns nothing.
-   */
   template <bool in_edges>
-  std::optional<edge_iterator> FindEdgeWithLabel(
-      GraphNode node, GraphNode key, const EdgeTy& data) const {
-    auto res = FindFirstOrLastEdgeWithLabel<
-        in_edges, FindEdgeWithLabelOptions::FindAny>(node, key, data);
-    if (!res) {
-      return std::nullopt;
+  class EdgeIteratorComparator {
+  public:
+    using Graph = LC_CSR_CSC_Graph<
+        NodeTy, EdgeTy, EdgeDataByValue, HasNoLockable, UseNumaAlloc,
+        HasOutOfLineLockable, FileEdgeTy>;
+    const Graph* graph_;
```

Comment:
Why not `const Graph&`?

Reply:
I just went with `*` since `this` was already a pointer type